crates/lair_keystore/src/store_sqlite.rs

```diff
@@ -680,7 +678,9 @@ mod tests {
         let mut sqlite = tmpdir.path().to_path_buf();
         sqlite.push("db.sqlite3");
 
-        let db_key = sodoken::BufReadSized::new_no_lock([0; 32]);
```

Comment:
This is test code so fine, but it illustrates nicely how confusing the "locked" vs "not locked" behavior can be here. The interface used to accept either even though the DB key should be protected as far as I can tell from the production code. Must more explicit now even if we're forcing creating a protected array of 0s